uportal-war/src/main/java/org/jasig/portal/layout/dlm/PLFIntegrator.java

```diff
@@ -277,4 +273,41 @@ static Element appendChild( Element plfChild,
         return copy;
     }
 
+     /*
+      * Nested Types
+      */
+
+     /**
+      * Helps {@link PLFIntegrator} track the number of {@link NodeInfo} objects
+      * created for a single layout.  There is some evidence in the community
+      * that there is a bug within the <code>mergePLFintoILF</code> process that
+      * can occur with certain data.  When triggered, it pulls the process into
+      * an infinite(?) loop, causing it to instantiate {@link NodeInfo} objects
+      * until the JVM runs out of memory.  This class tracks the number these
+      * objects created and fails noisily when the specific threshold is
+      * crossed.
+      */
+     /* package-private */ static final class NodeInfoTracker {
```

Comment:
In this case I think your view is prudent.  NodeInfoTracker objects are referenced by and manipulated by 2 classes in the package.  I often like static inner classes for small details that are known only to the class in which they reside.

Done.